rsync_tmbackup.sh

```diff
@@ -250,12 +250,24 @@ fn_test_file_exists_src() {
 	fn_run_cmd_src "test -e '$1'"
 }
 
+# Outputs 0 if given directory is not a type of FAT filesystem.
+# Outputs greater than 0 otherwise. Has to be implemented for different OSes.
 fn_df_t_src() {
-	fn_run_cmd_src "df -T '${1}'"
+	OSTYPE_STR="$(fn_run_cmd_src 'echo ${OSTYPE}')"
+	case "${OSTYPE_STR}" in
+		linux*) fn_run_cmd_src "df -T '${1}' | awk 'NR==2' | awk '{print \$2}' | grep -c -i -e 'fat'" ;;
+		darwin*) fn_run_cmd_src "df -T msdos,exfat ${1} | awk 'NR==2' | grep -c -e '${1%/}'" ;;
```

Comment:
Looks like `-T ...` on darwin is the same as `-t ...` on linux: limit to listed file system
Does darwin always return the file system type? Then simply give no -T argument (`df '$1' |`...).

And same with grep+awk as on last name.

Reply:
I do not have a Mac, so I am not sure about this. I made the changes based on feedback from @kapitainsky , in #170 , but what I have written may not be the most efficient way of doing this. If anyone would like to test and improve other OS cases, I would be glad.

Thanks again for your thoughtful feedback, much appreciated.

Reply:
mac OSX `df` is not likly the best source of this info: https://www.unix.com/man-page/osx/1/df/

😕 the goal is to know the filetype not the freespace, right?

if you want the file type and can already check that the system is darwin, I recomend using something like `diskutil info` with some grep/awk magic (see https://www.unix.com/man-page/OSX/8/diskutil/)

HOPE THIS HELPS

Reply:
Actually in my opinion df is much better. It is part of original BSD macOS heritage and will be there as long as macOS won't do some revolution. Diskutil  was added later by Apple - its output may different on different macOS versions (people still use old OSX systems - I know few) .  Just look at https://en.wikipedia.org/wiki/Disk_Utility - they talk about gui one but cmd version is just underlying engine.

Testing hell... when you can safely assume that df works always the same.

On top of it df usage results with one liner:

`df -T msdos,exfat ${1} | awk 'NR==2' | grep -c -e '${1%/}'`

very clear what it does (you want to check if filesystem in question is msdos or exfat) which means much less error prone. 

going for diskutil path will result with much more complex code. I don't think that "grep/awk magic" is always the best answer:)

Reply:
🤔 All good points.
...TL;DR; I've been known to think the preverbal space-shuttle solution when others just want a golf-cart so they can get around. I was thinking about the "filesystem type checks" (space shuttle) more than the "compatibility" (golf cart)

----

🤓 As for the:
> Testing hell... when you can safely assume that df works always the same.

safe or not _ANY_ assumption in code should be tested ... `df -T` behavior something I'm happy to take on as part of the testing I'm advocating for over in the testing threads.

Conclusion: I guess`df` works just fine for me.